testing/test-utils/src/lib/utils/nx-plugin.ts

```diff
@@ -0,0 +1,47 @@
+import { CreateNodes, CreateNodesContext, CreateNodesResult } from '@nx/devkit';
+import { vol } from 'memfs';
+import { MEMFS_VOLUME } from '../constants';
+
+/**
+ * Unit Testing helper for the createNodes function of a Nx plugin.
+ * This function will create files over `memfs` from testCfg.matchingFilesData
+ * and invoke the createNodes function on each of the files provided including potential createNodeOptions.
+ * It will aggregate the results of each invocation and return the CreateNodesResult.
+ *
+ * @example
+ * ```ts
+ * const {projects = {}} = await createFilesAndInvokeCreateNodesOnThem(createNodes, context, undefined, { matchingFilesData});
+ *  // project should have one target created
+ *  const targets = projects[projectRoot]?.targets ?? {};
+ *  expect(Object.keys(targets)).toHaveLength(1);
+ *  // target should be the init target
+ *  expect(targets[`${CP_TARGET_NAME}--init`]).toBeDefined();
+ *  ```
+ *
+ * @param createNodes
+ * @param context
+ * @param createNodeOptions
+ * @param mockData
+ */
+export async function createFilesAndInvokeCreateNodesOnThem<
+  T extends Record<string, unknown> | undefined,
+>(
+  createNodes: CreateNodes,
+  context: CreateNodesContext,
+  createNodeOptions: T,
+  mockData: {
+    matchingFilesData: Record<string, string>;
+  },
+): Promise<CreateNodesResult> {
+  const { matchingFilesData } = mockData;
+  vol.reset();
+  vol.fromJSON(matchingFilesData, MEMFS_VOLUME);
```

Comment:
I was not sure if this place is the best as it is not 100% clear to the user of this function that memfs is used. Also a cleanup in `beforeEach` for `vol.reset()` is a thing i avoided as it splits the code more.. 

Any feedback welcome

Reply:
I would not call vol.reset before (what if the caller does preprocessing) or after (that would remove all the created files in this function). I would have the caller take care of it and make sure all is set up for the function, e.g. mocking setup.

I would also call this function something to indicate it works on virtual system not a regular fs.

Reply:
removed the reset call.